common/lib/dependabot/metadata_finders/base/changelog_finder.rb

```diff
@@ -346,7 +347,14 @@ def gitlab_client
 
         def github_client
           @github_client ||= Dependabot::Clients::GithubWithRetries.
-                             for_github_dot_com(credentials: credentials)
+                             for_source(source: source, credentials: credentials)
+        end
+
+        def github_client_for_source(client_source)
```

Comment:
Maybe add a code comment to this method explaining that in addition to cloud, GHES also uses the GitHub client... because if I didn't have the context of this PR, I'd probably forget about GHES and wonder why we need to specify a source for GitHub, since I'm so used to thinking primarily of cloud.

Reply:
I'd like to eventually remove all the usages of `Dependabot::Clients::GithubWithRetries.for_github_dot_com(credentials: credentials)` as there's only a few left after this and that would make it harder to ignore the source.

We should support source for the other clients as well but I'm not well equipped to test them all out.